op-e2e/actions/interop/dsl/assertions.go

```diff
@@ -11,3 +17,156 @@ func RequireL2UnsafeNumberEquals(t require.TestingT, c *Chain, unsafeNumber uint
+		cfg.updateExpectedCrossUnsafe = true
+	}
+}
+
+func WithUpdateLocalSafe() func(cfg *updateExpectedConfig) {
+	return func(cfg *updateExpectedConfig) {
+		cfg.updateExpectedLocalSafe = true
+	}
+}
+
+func WithUpdateSafe() func(cfg *updateExpectedConfig) {
+	return func(cfg *updateExpectedConfig) {
+		cfg.updateExpectedSafe = true
+	}
+}
+
+func WithUpdateFinalized() func(cfg *updateExpectedConfig) {
+	return func(cfg *updateExpectedConfig) {
+		cfg.updateExpectedFinalized = true
+	}
+}
+
+func RequireSupervisorChainHeads(t helpers.Testing, super *SupervisorActor, chain *Chain, unsafe, crossUnsafe, localSafe, safe, finalized eth.BlockID) {
+	status, err := super.SyncStatus(t.Ctx())
+	require.NoError(t, err)
+
+	cStatus, ok := status.Chains[chain.ChainID]
+	require.True(t, ok, "chain id not present in supervisor sync status %s", chain.ChainID)
+	passed := true
+	passed = passed && assert.Equal(t, unsafe, cStatus.LocalUnsafe.ID(), "unsafe for chain id: %s", chain.ChainID)
+	passed = passed && assert.Equal(t, crossUnsafe, cStatus.CrossUnsafe, "cross unsafe for chain id: %s", chain.ChainID)
+	passed = passed && assert.Equal(t, localSafe, cStatus.LocalSafe, "local safe for chain id: %s", chain.ChainID)
+	passed = passed && assert.Equal(t, safe, cStatus.CrossSafe, "safe for chain id: %s", chain.ChainID)
+	passed = passed && assert.Equal(t, finalized, cStatus.Finalized, "finalized for chain id: %s", chain.ChainID)
+	require.True(t, passed, "One or more supervisor chain heads assertions failed")
+}
+
+func RequireUnsafeTimeOffset(t helpers.Testing, c *Chain, timeOffset uint64) {
+	require.Equal(t, timeOffset, c.Sequencer.L2Unsafe().Time-c.RollupCfg.Genesis.L2Time)
+}
+
+func RequireL1Heads(t helpers.Testing, system *InteropDSL, latest, finalized uint64) {
+	safeHeader, err := system.Actors.L1Miner.EthClient().BlockByNumber(t.Ctx(), big.NewInt(int64(rpc.LatestBlockNumber)))
+	require.NoError(t, err)
+	require.Equal(t, latest, safeHeader.Number().Uint64(), "L1 latest number does not match expectation")
+
+	finalizedHeader, err := system.Actors.L1Miner.EthClient().BlockByNumber(t.Ctx(), big.NewInt(int64(rpc.FinalizedBlockNumber)))
+
+	var finalizedActual uint64
+	if err != nil {
+		if err.Error() != "finalized block not found" {
+			require.NoError(t, err)
+		}
+		finalizedActual = 0
+	} else {
+		finalizedActual = finalizedHeader.Number().Uint64()
+	}
+
+	require.Equal(t, finalized, finalizedActual, "L1 finalized number does not match expectation")
+}
+
+type SyncStatusAsserter struct {
+	t         helpers.Testing
+	system    *InteropDSL
+	chain     *Chain
+	ExpStatus *eth.SyncStatus
+}
+
+func NewSyncStatusAsserter(t helpers.Testing, system *InteropDSL, chain *Chain) *SyncStatusAsserter {
+	return &SyncStatusAsserter{
+		t:         t,
+		system:    system,
+		chain:     chain,
+		ExpStatus: chain.Sequencer.SyncStatus(),
+	}
+}
+
+func (a *SyncStatusAsserter) RequireSeqSyncStatus(options ...updateExpectedOption) {
```

Comment:
My understanding of this DSL is that it stores initial L2CL sync status when initialization, and when we check fetch the current L2CL sync status, and check each head is advanced by checking that it matched the unsafe head. It would be nice to add the behavior in the godoc string.